### PR TITLE
add unit tests for tags.scm queries

### PR DESCRIPTION
**Pull request by the-mikedavis**
tree-sitter v0.20.6 was just published and it adds a [testing mechanism for tags queries](https://github.com/tree-sitter/tree-sitter/pull/1547) very similar to how the highlights tests work, except it's `test/tags/*` instead of `test/highlight/*`.

So here I have a fixture module from Mint with the tags tests. Should we add some more tests? I can't think of any edge-cases off the top of my head that would be interesting to test, but the more the merrier IMO 😄

### CODE REVIEW

**Comment by the-mikedavis** [comment]
looks like the CI failure is coming from [here](https://github.com/livebook-dev/livebook/blob/fe53c6a85265a51b59bce280d540b02b6f4c103e/app_builder/lib/app_builder/windows.ex#L165)

File: test/tags/mix.exs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# https://github.com/elixir-mint/mint/blob/e7199d1163e5a8a1b0001cacc93063e866dbaa35/mix.exs
+defmodule Mint.MixProject do
```
Comment (jonatanklosko) [inline test/tags/mix.exs:2]: I think it would make more sense to come up with minimal examples matching the queries in tags.scm. Something like [here](https://github.com/elixir-lang/tree-sitter-elixir/blob/a11a686303355a518b0a45dea7c77c5eebb5ec22/test/highlight/module.ex), this way we avoid duplication and if something fails it should be more clear. We can keep this file under `tags/integration` if you like :)

From edge cases, we are missing `defimpl`, function call as in `|> foo` and maybe another definition keyword like `defmacro`.

**Review by jonatanklosko — COMMENTED** [review]
Hey @the-mikedavis, fantastic news, great work on improving tree-sitter!! Just one comment regarding the test :)

The integration tests detected a new parsing error, and the code you referenced is clearly mis--highlighted because of that. I will have a look a this!

File: test/tags/mix.exs (hunk 1 above)
Comment (the-mikedavis) [reply test/tags/mix.exs]: Welp I found a bug in my PR :sweat_smile:

Looks like you can't have directories under `test/{highlight,tags}` - it fails trying to read the extension to determine which parser to use. I'll set myself a reminder to submit a patch for that! In the meantime I've removed `test/tags/mix.exs`. I think the new module.ex tests cover everything we saw in `mix.exs` in a more compact way.

File: src/tree_sitter/parser.h [hunk 2]
```diff
@@ -123,6 +123,7 @@ struct TSLanguage {
     unsigned (*serialize)(void *, char *);
     void (*deserialize)(void *, const char *, unsigned);
   } external_scanner;
+  const TSStateId *primary_state_ids;
```
Comment (jonatanklosko) [inline src/tree_sitter/parser.h:126]: Will be generated :)

```suggestion
```

**Review by jonatanklosko — APPROVED** [review]
LGTM!

The CI is fixed in #28, so feel free to merge afterwards :)
